IPython/core/debugger.py

```diff
@@ -280,27 +280,35 @@ def __init__(self, color_scheme=None, completekey=None,
 
         # Set the prompt - the default prompt is '(Pdb)'
         self.prompt = prompt
+        self.skip_hidden = True
 
     def set_colors(self, scheme):
         """Shorthand access to the color table scheme selector method."""
         self.color_scheme_table.set_active_scheme(scheme)
         self.parser.style = scheme
 
+    def stop_here(self, frame):
+        """
+        Stop here is usually used only to know wether commands should stop on this frame,
+        we are also going to use it to hide frames in where, up and down for example.
+
+        This will also imply that many other commands will now skip hidden tracebacks.
+        """
+
+        return super().stop_here(frame) and not frame.f_locals.get("__tracebackhide__", False)
```

Comment:
Should stop_here() be:
return super().stop_here(frame) and not self.should_skip(frame)

I don't think that if `super().stop_here(frame)` is false it means the frame should be hidden.

then `should_skip` would be:
`return frame.f_locals.get("__tracebackhide__", False)`

Reply:
and:
```
def should_hide(self, frame):
    return self.should_skip(frame)
```

Reply:
That's fair, updated to reflect your suggestion.